homeassistant/components/twinkly/light.py

```diff
@@ -98,7 +103,7 @@ def __init__(
         self._attr_available = False
         self._current_movie: dict[Any, Any] = {}
         self._movies: list[Any] = []
-        self._software_version = ""
+        self._software_version = device_info.get(ATTR_SW_VERSION)
```

Comment:
```suggestion
        self._software_version = device_info[ATTR_SW_VERSION]
```

Reply:
The question is though, should this be mangled into this `device_info`? As that is the result of `client.get_details()`.

I feel like this should just be passed separately, instead of modifying an upstream response.

Reply:
Alternatively, we could query this on setting up the entry on integration init, and store this information in hass data (just like the client and device info)

Reply:
It could be just a naming confusion, as we want sw_version in the DeviceInfo of the entity.

Reply:
When testing it now, I think it was to be able to pass it to the init of the LightEntity.
self.hass is not available on init, but sw_version could be added as another parameter.

Reply:
I _think_ I did what you requested.